server/reviewsController.js

```diff
@@ -0,0 +1,192 @@
+/* eslint-disable no-restricted-syntax */
+/* eslint-disable guard-for-in */
+/* eslint-disable no-shadow */
+/* eslint-disable no-plusplus */
+/* eslint-disable no-console */
+const { db } = require('../database/index');
+
+const reviews = (req, res) => {
+  const { productId } = req.params;
+
+  const query = `
+  SELECT
+    reviews.id AS reviews_id, reviews.rating, reviews.summary, reviews.recommend, reviews.response, reviews.body, to_timestamp(reviews.date / 1000) AS date, reviews.reviewer_name, reviews.helpfulness, reviews_characteristics.value AS characteristics_value, reviews_characteristics.characteristic_id AS characteristics_id, characteristics.name AS characteristics_name, products.name AS product_name,
+    CASE WHEN EXISTS (SELECT reviews_photos.url FROM reviews_photos WHERE reviews_photos.review_id = reviews.id)
+      THEN (SELECT reviews_photos.url FROM reviews_photos WHERE reviews_photos.review_id = reviews.id)
+      ELSE ''
+    END AS photo_url,
+    CASE WHEN EXISTS (SELECT reviews_photos.id FROM reviews_photos WHERE reviews_photos.review_id = reviews.id)
+      THEN (SELECT id FROM reviews_photos WHERE reviews_photos.review_id = reviews.id)
+      ELSE NULL
+    END AS photo_id
+  FROM reviews, reviews_characteristics, characteristics, products
+  WHERE reviews.product_id = $1
+  AND reviews_characteristics.review_id = reviews.id
+  AND characteristics.id = reviews_characteristics.characteristic_id
+  AND reviews.product_id = products.id
+  GROUP BY reviews.id, reviews_characteristics.value, characteristics.name, products.name, reviews_characteristics.characteristic_id
+  ;`;
+
+  const output = {
+    product: productId,
+    productName: '',
+    meta: {},
+    results: [],
+  };
+
+  db.any(query, productId)
+    .then((result) => {
+      output.productName = result[0].product_name;
+
+      const reviewsTracker = {};
+
+      for (let i = 0; i < result.length; i++) {
```

Comment:
just minor feedback - you could make line count on this file a little smaller by outsourcing all your helpers =D. either way i dont think it matters tho!